### PR TITLE
Week 6 raul marquez

### PR DESCRIPTION
**Pull request by Echo5Mike**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -10,42 +9,164 @@
...
+// const people = [
+//   { name: "ford prefect", occupation: "a hitchhiker" },
+//   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+//   { name: "arthur dent", occupation: "a radio employee" }
+// ]
+// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 // b) Create the function that makes the test pass.
 
 
+// ReferenceError: occuPerson is not defined
```
Comment (elysemontano) [inline code-challenges.test.js:50]: Awesome job practicing TDD!

File: code-challenges.test.js [hunk 2]
```diff
@@ -10,42 +9,164 @@
...
+  array.forEach(value =>{  
+    //find a way to separate the first name and last name
+    let lName = value.name.charAt(value.name.indexOf(' ')+1).toUpperCase() + value.name.substring(value.name.indexOf(' ')+2)
+
+    let wholeName = (value.name.charAt(0).toUpperCase() + value.name.substring(1,(value.name.indexOf(' '))) +" "+ lName)
+
+    endGame.push(`${wholeName} is ${value.occupation}.`)
+  })
+  return endGame
+}
```
Comment (elysemontano) [inline code-challenges.test.js:67]: Good job!  Can you think of some other approaches to achieve the same result that can account for edge cases too?  For instance, would this work if Arthur had no last name or a person had three names?

File: code-challenges.test.js [hunk 3]
```diff
@@ -10,42 +9,164 @@
...
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+describe("justNums", () => {
+  const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
```
Comment (elysemontano) [inline code-challenges.test.js:92]: Test looks good!  I would suggest keeping the variables inside the scope you are working with.  In this case, the variables are being used inside the "it" block.

File: code-challenges.test.js [hunk 4]
```diff
@@ -10,42 +9,164 @@
...
+      //how are we going to get the remainder only?
+
+const justNums = (array) => {
+  numsDiv3=[]
+  array.filter(value =>{
+    if (typeof value === "number")
+    numsDiv3.push(value%3)
+  })
+return numsDiv3
+}
```
Comment (elysemontano) [inline code-challenges.test.js:120]: Logic looks great!

File: code-challenges.test.js [hunk 5]
```diff
@@ -10,42 +9,164 @@
...
+//this is not ruby so I might have to forLoop to add the number back up...
+// const cubedSum = (array)=>{
+
+const cubedSum = (array)=>{
+  temp=[]
+
+  // try math.sum(args)
+  for (let i = 0; i < array.length; i++) {
+      temp.push((array[i]**3))  
+    }
```
Comment (elysemontano) [inline code-challenges.test.js:170]: Logic looks great!  It definitely can be challenging to switch our brain from Ruby back to Javascript.  Can you think of a way to do the same thing with a higher-order function rather than a for loop?

File: interview-questions.md [hunk 6]
```diff
@@ -8,43 +8,67 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
   Your answer:
+  It is my understanding you would have to create a new column and name it the same as the foreign key that you want to pass it.
 
   Researched answer:
-
+It is my understanding you would have to create a new column and name it the same as the foreign key that you want to pass it. You would also need to use the convention of has_many__?__.  Once you updated the model Cohort  you would also need to update the foreign key of the Student model.
```
Comment (elysemontano) [inline interview-questions.md:14]: Correct, the foreign key is on the Student model.  You certainly would want to check to make sure that both models are set up with the proper associations.  To update the Student model to have the foreign key, you would want to run a migration to update the schema to have the added column.

File: interview-questions.md [hunk 7]
```diff
@@ -8,43 +8,67 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 action: "GET"    route: /students/2/edit    
+Edit = if you wanted to get and edit done you would need the edit controller method
 
-action: "PATCH"  route: /students/2      
+action: "PATCH"  route: /students/2     
+Update = both patch and put will need the update controller method if you are "updating" a particular student.
 
 action: "DELETE" route: /students/2      
+Destroy = is the method used to delete a particular student.
```
Comment (elysemontano) [inline interview-questions.md:60]: Awesome job!

File: interview-questions.md [hunk 8]
```diff
@@ -8,43 +8,67 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+1. As a user I see a page with a summary of the website.
+2. As a user i see a list of "chores" or things todo.
+3. next to each todo item i can see the progress of each todo item.  not started, in progress or completed.
+4. As a user I can add a new todo item to the list. 
+5. As a user i can update my todo from not started to in progress
+6. As a user I can update my todo from in progress to completed.
+7. As a user I see all my completed todo's at the bottom of the list.
+8. As a user I can see a completed task button where I can see all my completed tasks.
+9. As a user I can sign in with a user name and password.
+10. As a user I can share my profile with others.
```
Comment (elysemontano) [inline interview-questions.md:73]: Love that you added your own stretch challenge and also user login!  You have most of the CRUD actions in your stories as well.  Could the user want to delete a task as well?

File: rails-commenting.rb [hunk 9]
```diff
@@ -33,14 +33,14 @@ def create
...
-  # ---6)
+  # ---6)we are calling our def "edit" and will be used as method to edit our blog post
   def edit
     @post = BlogPost.find(params[:id])
   end
 
   def update
     @post = BlogPost.find(params[:id])
-    # ---7)
+    # ---7)@post is assigned to blog via id and it is being updated in the database with the .update method.
```
Comment (elysemontano) [inline rails-commenting.rb:43]: Correct!  What is (blog_post_params) doing though?

File: rails-commenting.rb [hunk 10]
```diff
@@ -54,15 +54,15 @@ def destroy
...
 
-  # ---9)
+  # ---9) private add restriction to the params/method that you do not want you user to have access to.  Everything under private will have those restrictions. 
   private
   def blog_post_params
-    # ---10)
+    # ---10)It is my understanding that .require is requireing that the blog post contain the two permited keys.
     params.require(:blog_post).permit(:title, :content)
   end
 end
```
Comment (elysemontano) [inline rails-commenting.rb:68]: Awesome work on this section!

**Review by elysemontano — APPROVED** [review]
Awesome work Raul!  I can tell you put a lot of work into this assessment!  When answering interview questions, try to address all parts of the question even if you are not entirely sure.  Acknowledging the initial question(s) in their entirety is a big first step and can help drive forward the answer.  Keep up the great work!
